### PR TITLE
mp sync params & grads & opt states.

### PR DESCRIPTION
**Pull request by wuhuachaocoding**
### PR types

Others
### PR changes

Others
### Describe

mp sync params & grads & opt states for embedding、bias and layernorm.
- sync grad before opt step
- sync param after opt step
- sync master param after opt step
- sync opt state after opt step

see: https://github.com/PaddlePaddle/Paddle/pull/50676

### CODE REVIEW

**Review by gongweibao — DISMISSED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_mp_model.py [hunk 1]
```diff
@@ -181,6 +181,145 @@ def forward(self, x):
...
+        self.model_parallel_size = 2
+        self.data_parallel_size = 1
+        strategy.hybrid_configs = {
+            "dp_degree": self.data_parallel_size,
+            "mp_degree": self.model_parallel_size,
+            "pp_degree": 1,
+        }
+        strategy.mp_configs = {
+            "sync_param": False,
+            "sync_grad": False,
```
Comment (gongweibao) [inline python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_mp_model.py]: sync_grad和sync_moment是指？

**Review by gongweibao — CHANGES_REQUESTED** [review]

**Review by ForFishes — APPROVED** [review]
LGTM

File: python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_mp_model.py (hunk 1 above)
Comment (wuhuachaocoding) [reply python/paddle/fluid/tests/unittests/collective/fleet/hybrid_parallel_mp_model.py]: 一个标志：代表是否同步grad, 是否同步moment. 和静态图对齐。

**Review by gongweibao — APPROVED** [review]
LGTM

**Pull request merged**